### ISSUE TITLE
Request for configs for valid spawn blocks for structures 

### ISSUE THREAD

**Issue by bookerthegeek** [Azanor/thaumcraft#1751]
As the title states, It would be nice to have a config option to specify which blocks are considered valid for spawning of structures for mod packs that convert the world into blocks that are not on the list currently in code as valid spawn-able blocks.

Once again thank you for the wonderful Mod you have created and have a wonderful day!

**Comment by MalkContent** [comment]
Could you split up that request in multiple sentences? I have no idea what the toots you are talking about, and maybe that helps.

**Comment by bookerthegeek** [comment]
* As of now, Thaumcraft hard-codes the blocks that it is to consider "acceptable" for spawning in structures. Such as the:
 * Dungeon
 * Obelisk
 * Hilltop Stones
 * Greatwood Trees
* It would be nice for packs creators to be able to specify another block for those to be able to spawn on. Such as:
  * Regrowth, covers the world in cracked sand

I am not asking that you add all the possible blocks to the list of acceptable spawn. Just a config option for users to add an additional block to the "This is a valid block for World gen to spawn on" List.

Sorry if my previous post was not very well formatted, as I was on my phone at the time. Hope this helps. :)

**Comment by MalkContent** [comment]
Nice to finally meet someone who can fuck up words as good as myself ^^  

If I understand you correctly: You want a config option to allow thaumcraft structures to spawn in/on additional blocks, i.e. mod blocks that are similar to the vanilla blocks which the structures currently spawn in/on.

**Comment by bookerthegeek** [comment]
Exactly. This is mainly an issue on modpacks that cover world in a single block type (i.e.regrowth). But this would also be usefull for mods that add biome as some biome are covered in non spawnable blocks as well.

I know there is another mod that attempted to add this as a config but it did not work. https://github.com/BluSunrize/WitchingGadgets/issues/46 in case you were wondering.
